src/ModelMerger.py

```diff
@@ -0,0 +1,158 @@
+
+        This function takes a list of metabolite IDs from namespace 2 and attempts to translate each
+        to the best matching metabolite ID in namespace 1 based on a matching score. The function ensures
+        that no two metabolites from the to_translate list are translated to the same ID in namespace 1.
+        If two or more metabolites have the same best match, only the one with the highest score
+        (or the first alphanumerically in case of a tie) is assigned. Unassigned metabolites will retain
+        their original IDs.
+
+        Parameters:
+            to_translate (list): A list of metabolites that need to be translated.
+            matches_df (DataFrame): A DataFrame containing the potential matches between namespaces,
+                                    with columns ['source_namespace', 'target_namespace', score_type].
+            score_type (string): A string indicating which column from matches_df contains the matching
+                                 score that will be used to rank the metabolite matches.
+        """
+        # # Sort the metabolites list alphanumerically by their ID
+        # to_translate = sorted(to_translate, key=lambda metabolite: metabolite.id)
+
+        # Remove "_t" suffix from the metabolite IDs
+        for met in to_translate:
+            met.id = re.sub(r"_t$", "", met.id)
+
+        sorted_matches = matches_df.sort_values(by=[score_type, 'source_namespace', 'target_namespace'],
+                                                ascending=[False, True, True])
+
+        best_matches = dict()
+
+        # Iterate over the to_translate list
+        for met_id in sorted_matches["source_namespace"]:
+            if met_id in [m.id for m in to_translate]:
+                #print([m.id for m in to_translate])
+                #print(met_id)
+                # Filter the matches for the current metabolite
+                matches_for_met = sorted_matches[sorted_matches['source_namespace'] == met_id]
+                # Assign the best available match
+                for _, row in matches_for_met.iterrows():
+                    match_id = row['target_namespace']
+                    if (met_id not in best_matches.keys()) and (match_id not in best_matches.values()):
+                        best_matches[met_id] = match_id
+                        #print(met_id + ": " + match_id + ", " + str(row[score_type]))
+                #else:
+                #    # If no match is found, keep the original ID
+                #    best_matches[met_id] = met_id
+                #    print(met_id + ": " + met_id)
+
+        # apply the id translation to the metabolite and re-appends the translation compartment suffix
+        for met in to_translate:
+            try:
+                met.id = best_matches[met.id] + "_t"
+            except KeyError:
+                met.id = met.id + "_t"
+
+    def translate_ids(self, score_thr, score_type="Name_score"):
```

Comment:
default should be "total_score" not "Name_score"

Reply:
We should remove the thresholding completely. We should not implement that here, but somewhere before. Here you can assume that you get a clean list of met1 to met2, which is also contains only unique matches. 

This should be a separate issue.